zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -582,6 +582,13 @@
       }
       websocketMsgSrv.saveInterpreterBindings($scope.note.id, selectedSettingIds);
       console.log('Interpreter bindings %o saved', selectedSettingIds);
+
+      _.forEach($scope.note.paragraphs, function(n, key) {
+        if (n.text && !n.text.startsWith('%')) {
```

Comment:
Thanks for reviewing!
If user defined `%foo` explicitly in paragraph, we don't need to change `editorSetting` cause `editorSetting` such as `language`, `editOnDblClick` of this paragraph will be same even if the default interpreter changed. (This field is saved in `note.json` file when user type `%foo`.)
The changes will be only needed to those without `%foo` specified.